### DISCUSSION TITLE
Web browsing tool

### DISCUSSION THREAD

**Discussion by onomic** [TransformerOptimus/SuperAGI#214] [Q&A] [answered]
Are there any web browsing tools?

**Comment by jakderrida** [comment]
Yeah, that's the reason you get the google keys. Although, if you're asking if it can browse and navigate, let's say like a university website for example, it'll likely struggle a bit. I poured money into Auto-GPT just so it could keep promising me it's studying to learn everything about navigating a Selenium browser (by my instructions) and how to analyze html and fetch through xpath (as I told it to) and it would just go on for at least two hours and accomplish absolutely nothing. Even when I specifically told Auto-GPT to just scrape all four links found in the body portion of a the link I also specified, it would make no progress. Eventually, I even put the four links in the task and it still couldn't return them to me.

**Comment by onomic** [comment]
Thanks, for Auto-GPT I found success with putting URLS in a text file and telling it to read the text file in the goals.

**Comment by neelayan7** [comment, marked as answer]
We have Google Search Tool as well as Web Scraper tool which can help you browse the internet as well as scrape the internet.

**Comment by BuyWhere** [comment]
For a production-ready data source, BuyWhere MCP provides structured product data (prices, availability, categories) that you could wire into your agent toolset. There is a builder challenge at buywhere.ai/challenge if you are exploring commerce agent use cases.
